server/src/main/java/com/saralein/server/middleware/AuthMiddleware.java

```diff
@@ -0,0 +1,63 @@
+package com.saralein.server.middleware;
+
+import com.saralein.server.request.Request;
+import com.saralein.server.response.ErrorResponse;
+import com.saralein.server.response.Response;
+import com.saralein.server.router.Routes;
+import java.util.Base64;
+
+public class AuthMiddleware implements Middleware {
+    private final Routes routes;
+    private final String username;
+    private final String password;
+    private final String realm;
+    private Callable next;
+
+    public AuthMiddleware(Routes routes, String username, String password, String realm) {
```

Comment:
I think this class is probably doing too much. I feel like that actual authentication should be handled in another object, and the middleware should use that. Yeah?